storage/buckets/buckets_test.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buckets
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+func TestCreate(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	bucketName := tc.ProjectID + "-storage-buckets-tests"
+
+	// Clean up bucket before running tests.
+	deleteBucket(ioutil.Discard, bucketName)
+	if err := createBucket(ioutil.Discard, tc.ProjectID, bucketName); err != nil {
+		t.Fatalf("createBucket: %v", err)
+	}
+}
+
+func TestCreateBucketClassLocation(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	name := tc.ProjectID + "-storage-buckets-tests-attrs"
+
+	// Clean up bucket before running the test.
+	deleteBucket(ioutil.Discard, name)
+	if err := createBucketClassLocation(ioutil.Discard, tc.ProjectID, name); err != nil {
+		t.Fatalf("createBucketClassLocation: %v", err)
+	}
+	if err := deleteBucket(ioutil.Discard, name); err != nil {
+		t.Fatalf("deleteBucket: %v", err)
+	}
+}
+
+func TestListBuckets(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	bucketName := tc.ProjectID + "-storage-buckets-tests"
+
+	buckets, err := listBuckets(ioutil.Discard, tc.ProjectID)
+	if err != nil {
+		t.Fatalf("listBuckets: %v", err)
+	}
+
+	var ok bool
+outer:
+	for attempt := 0; attempt < 5; attempt++ { // for eventual consistency
```

Comment:
Please use `testutil.Retry`. Be sure to fail with `r.Errorf`, rather than `t` directly.